azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -207,30 +203,23 @@ def _fetch_vm_settings(self, force_update=False):
         vm_settings, vm_settings_updated = (None, False)
 
         if conf.get_enable_fast_track():
-            def save_to_history(etag, text):
-                # The vmSettings are updated independently of the WireServer goal state and they are saved to a separate directory
-                history = GoalStateHistory(datetime.datetime.utcnow().isoformat(), etag)
-                history.save_vm_settings(text)
-
             try:
                 vm_settings, vm_settings_updated = self._wire_client.get_host_plugin().fetch_vm_settings(force_update=force_update)
 
             except VmSettingsNotSupported:
                 pass
             except VmSettingsError as exception:
-                save_to_history(exception.etag, exception.vm_settings_text)
+                # ensure we save the vmSettings if there were parsing errors
+                self._history.save_vm_settings(ExtensionsGoalStateFromVmSettings.redact(exception.vm_settings_text))
                 raise
             except ResourceGoneError:
                 # retry after refreshing the HostGAPlugin
                 GoalState.update_host_plugin_headers(self._wire_client)
                 vm_settings, vm_settings_updated = self._wire_client.get_host_plugin().fetch_vm_settings(force_update=force_update)
 
-            if vm_settings_updated:
-                save_to_history(vm_settings.etag, vm_settings.get_redacted_text())
-
-        return vm_settings
+        return vm_settings, vm_settings_updated
 
-    def _fetch_extended_goal_state(self, xml_text, xml_doc, vm_settings):
+    def _fetch_extended_goal_state(self, xml_text, xml_doc, force_vm_settings_update=False):
```

Comment:
in init, I split the member variables in 2 blocks, which I called "basic" and "extended", then initialize_basic_properties initializes the former and fetch_extended_goal_state the latter (I did not name fetch_extended_goal_state "initialize_extended_properties" because there is a fetch operation in it and I want that to be explicit)